gnd/src/main/java/com/google/android/gnd/model/basemap/tile/Tile.java

```diff
@@ -4,25 +4,42 @@
 
 @AutoValue
 public abstract class Tile {
-    public enum State {
-        PENDING,
-        IN_PROGRESS,
-        DOWNLOADED,
-        FAILED
-    }
-
-    public abstract String getId();
-    public abstract String getPath();
-    public abstract State getState();
-
-    public static Builder newBuilder() {return new AutoValue_Tile.Builder();}
-
-    @AutoValue.Builder
-    public abstract static class Builder {
-        public abstract Builder setId(String id);
-        public abstract Builder setPath(String path);
-        public abstract Builder setState(State state);
-
-        public abstract Tile build();
-    }
-}
\ No newline at end of file
+  public enum State {
+    PENDING,
+    IN_PROGRESS,
+    DOWNLOADED,
+    FAILED
+  }
+
+  public abstract String getId();
+
+  public abstract String getPath();
+
+  public abstract State getState();
+
+  public static Builder newBuilder() {
+    return new AutoValue_Tile.Builder();
+  }
+
+  public static String pathFromId(String tileId) {
+    // Tile ids are stored as x-y-z. Paths must be z-x-y.mbtiles.
+    // TODO: Convert tile ids to paths in a less restrictive and less hacky manner.
+    String[] fields = tileId.replaceAll("[()]", "").split(", ");
+    String filename = fields[2] + "-" + fields[0] + "-" + fields[1];
```

Comment:
Aren't we keeping a mapping between ID and filename in the DB? Or is that still TBD?

Reply:
We are, but at the moment we never actually set the path for a given tile anywhere--this may not be the right place for this method, I wasn't sure where we should be doing this sort of work.

Our Ids are in X-Y-Z format, while our source tile files in the storage api are stored as Z-X-Y.mbtiles, so we need to do this transformation somewhere--would it be more appropriate to have it in the download worker, keeping the filename independent of the source file name? (at the moment, I'm just saving the files with the same name as they have in storage).

Reply:
I don't have the whole flow paged into memory anymore, but I assume this is because the TileProvider needs an easy way to resolve the filename of the tiles without necessarily going to the DB? If so I think this is fine, maybe just add a TODO to think about a better place for the method in the future (not sure where is best).

Reply:
Sounds good! Yep, once things settle down a bit and we get to cleaning some of this stuff up we can sit down and go over the flow and where precisely things should live. Made the todo more specfi, thanks!